### PR TITLE
Implement DistGitMRHandler.run() 

### PR DESCRIPTION
**Pull request by jpopelka**
While deployed to [stream-stg](https://console.pro-eu-west-1.openshift.com/console/project/stream-stg) this code created [this dist-git MR](https://gitlab.com/packit-service/rpms/open-vm-tools/-/merge_requests/6) (the patches there are removed because the maintainer did not remove them in [the latest commit](https://gitlab.com/packit-service/rpms/open-vm-tools/-/commit/8c349949521e5c3fcd5c1811d1acbc5a752b385e)) from [this source-git MR](https://gitlab.com/packit-service/src/open-vm-tools/-/merge_requests/5).

- [x] Update deployment playbooks
- [x] Make it work in the deployed service with the service's credentials
- [x] Link the two MRs
- [x] Tests
- [x] Containerize tests and run them in Zuul

### CODE REVIEW

File: hardly/handlers/distgit.py [hunk 1]
```diff
@@ -31,14 +31,46 @@ def __init__(
             job_config=job_config,
             event=event,
         )
+        self.mr_identifier = event.get("identifier")
+        self.mr_title = event.get("title")
+        self.mr_description = event.get("description")
+        # Gitlab specific, but hardly is too
+        self.mr_url = (
+            f"{event.get('project_url')}/-/merge_requests/{self.mr_identifier}"
```
Comment (mfocko) [inline hardly/handlers/distgit.py:39]: If you look at the MR webhook in test data in service, there is an `url` field in `event`. Or there is a problem with that?

File: hardly/handlers/distgit.py [hunk 2]
```diff
@@ -31,14 +31,46 @@ def __init__(
...
+        self.source_project_url = event.get("source_project_url")
+        self.target_repo_branch = event.get("target_repo_branch")
 
     def run(self) -> TaskResults:
         """
         If user creates a merge-request on the source-git repository,
         create a matching merge-request to the dist-git repository.
         """
+        if self.target_repo_branch != "c9s":
+            return TaskResults(success=True, details={})
```
Comment (mfocko) [inline hardly/handlers/distgit.py:50]: For the debugging purposes message would be nice :)

File: hardly/jobs.py [hunk 3]
```diff
@@ -36,11 +36,7 @@ def process_message(
                 logger.debug(f"{topic} not in {topics}")
                 return []
 
-        event_object: Any
-        if source == "centosmsg":
```
Comment (mfocko) [inline hardly/jobs.py:40]: oh i see now, i was wondering how come we don't need it :D

File: tests/data/webhooks/gitlab/mr_event.json [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    "source_branch": "DnD-RpcV3-OOB",
+    "source_project_id": 28555414,
+    "state_id": 1,
+    "target_branch": "c9s",
+    "target_project_id": 25495423,
+    "time_estimate": 0,
+    "title": "Yet another testing MR",
+    "updated_at": "2021-08-02 14:36:37 UTC",
+    "updated_by_id": null,
+    "url": "https://gitlab.com/packit-service/src/open-vm-tools/-/merge_requests/5",
```
Comment (mfocko) [inline tests/data/webhooks/gitlab/mr_event.json:57]: ^ this

File: hardly/handlers/distgit.py [hunk 5]
```diff
@@ -31,14 +31,46 @@ def __init__(
...
         )
-        return TaskResults(success=True, details={})
+
+        self.api = PackitAPI(
+            config=self.service_config,
+            package_config=self.package_config,
+            upstream_local_project=self.local_project,
+        )
+
+        dg_mr = self.api.sync_release(
```
Comment (csomh) [inline hardly/handlers/distgit.py:65]: We should use `update_dist_git` here.

Comment (jpopelka) [reply hardly/handlers/distgit.py:65]: We are, via `sync_release()`.
I initially also thought I'd use `update_dist_git` directly, but then I realized I'd basically be a copy of `sync_release()`.

File: hardly/handlers/distgit.py (hunk 1 above)
Comment (jpopelka) [reply hardly/handlers/distgit.py:39]: No problem, I was just lazy to create a PR in p-s :-)
Here it is packit/packit-service#1197

File: hardly/jobs.py (hunk 3 above)
Comment (jpopelka) [reply hardly/jobs.py:40]: This is a relic (copied from packit-service worker) from handling stream repositories on git.centos.org.
We're not going to handle those in hardly and it'll eventually be removed also from packit-service.

**Review by TomasTomecek — APPROVED** [review]
had a brief look

**Comment by mfocko** [comment]
> Pull request merge failed: Resource not accessible by integration

GitHub Workflow update I guess…

**Pull request merged**